vendor/cppbackport-master/lib/fs/rename.cpp

```diff
@@ -45,7 +45,17 @@ void rename(const path& from, const path& to)
 		if (copy_file(from, to))
 			cpp17::filesystem::remove(from);
 	}
-}
+} // end rename
+
+void rename(const path& from, const path& to, __attribute__((unused)) std::error_code& ec)
+{
+	if (std::rename(from.c_str(), to.c_str()) != 0)
```

Comment:
this is literally the same function as the other one. why not just call that, wrap it with try/catch and set error code if it throws?

Reply:
in case someone does want to implement std:error_code